core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -708,7 +709,7 @@ private[spark] class TaskSetManager(
       put(info.executorId, clock.getTime())
     sched.dagScheduler.taskEnded(tasks(index), reason, null, null, info, taskMetrics)
     addPendingTask(index)
-    if (!isZombie && state != TaskState.KILLED) {
+    if (!isZombie && state != TaskState.KILLED && !reason.isInstanceOf[TaskCommitDenied]) {
```

Comment:
maybe add a short comment here, something like
```
// Do not abort the task if its attempt to commit is denied because another
// attempt of the same task is expected to commit the result instead
```